scylla/src/transport/session.rs

```diff
@@ -489,23 +489,35 @@ impl Session {
         let known_nodes = config.known_nodes;
 
         #[cfg(feature = "cloud")]
-        let known_nodes = if let Some(cloud_servers) =
-            config.cloud_config.as_ref().map(|cloud_config| {
-                cloud_config
+        let cloud_known_nodes: Option<Vec<InternalKnownNode>> =
+            if let Some(cloud_config) = config.cloud_config {
+                let cloud_servers = cloud_config
                     .get_datacenters()
                     .iter()
                     .map(|(dc_name, dc_data)| {
-                        KnownNode::CloudEndpoint(CloudEndpoint {
+                        InternalKnownNode::CloudEndpoint(CloudEndpoint {
                             hostname: dc_data.get_server().to_owned(),
                             datacenter: dc_name.clone(),
                         })
                     })
-                    .collect()
-            }) {
-            cloud_servers
-        } else {
+                    .collect();
+                Some(cloud_servers)
+            } else {
+                None
+            };
+
+        #[cfg(not(feature = "cloud"))]
+        let cloud_known_nodes: Option<Vec<InternalKnownNode>> = None;
+
+        let known_nodes = cloud_known_nodes.unwrap_or_else(|| {
             known_nodes
-        };
+                .into_iter()
+                .map(|node| match node {
+                    KnownNode::Address(a) => InternalKnownNode::Address(a),
+                    KnownNode::Hostname(a) => InternalKnownNode::Hostname(a),
+                })
```

Comment:
You perform manual conversion, even though you implemented `From`. Why not use it then?

Reply:
Opps, I'll fix it.